go.mod

```diff
@@ -64,6 +65,7 @@ require (
 	github.com/DataDog/datadog-agent/pkg/otlp/model v0.33.0-rc.4
 	github.com/DataDog/datadog-agent/pkg/quantile v0.33.0-rc.4
 	github.com/DataDog/datadog-agent/pkg/security/secl v0.33.0-rc.4
+	github.com/DataDog/datadog-agent/pkg/tagset v0.33.0-rc.4
```

Comment:
We wanted to have this as a module so as to use it on the `pkg/otlp/model` module, to import that module on the OpenTelemetry Collector Datadog exporter. I personally don't want to import the `pkg/tagset` module directly from outside this repo. so we could make this module ['internal'](https://dave.cheney.net/2019/10/06/use-internal-packages-to-reduce-your-public-api-surface) for now. 

To be more explicit: if we make the path `pkg/internal/tagset` it would still fit the otlp use case (since it would be a separate module), but it would only be directly importable from modules/packages inside the datadog-agent repo (more concretely, modules/packages inside the `pkg` folder).

Reply:
I think `pkg/internal` wouldn't work, as it'd be inaccessible from `cmd/`?

Anyway, since we don't use `internal` anywhere else in the codebase, I'd prefer not to introduce the idea with this project.  In particular, I think that use of `internal` is probably the right way to solve our concerns around other people linking to our modules in hopes of a stable API.  Which means probably almost all of `pkg` and `cmd` should move under `internal`, with the exception of things that we *do* support externally.

Also, I've designed pkg/tagset to have a stable API, so I would lean toward supporting its use externally.

In other words, I think `internal` is a bigger conversation.